### ISSUE TITLE
Updated from v1.0 to v2.0 and now detached: true always throws a TypeError

### ISSUE THREAD

**Issue by rynz** [sindresorhus/execa#359]
Node: v12.4.0 and Electron v6.0.1
```
const { spawn } = require('child_process')
spawn('C:\\Users\\ryanj\\AppData\\Local\\HideAway\\Update.exe', [ '--createShortcut', 'HideAway.exe' ], { detached: true })
```
Works no worries.
```
const execa = require('execa')

const run = async () => {
  const { stdout } = await execa('C:\\Users\\ryanj\\AppData\\Local\\HideAway\\Update.exe', [ '--createShortcut', 'HideAway.exe' ], { detached: true })
  return stdout
}

run().catch(err => {
  const stack = err.stack ? err.stack : String(err)
  const message = `Run exception:\n${stack}`
  console.error(message)
})
```
Always throw the following error:
```
Run exception:
TypeError: Cannot destructure property `error` of 'undefined' or 'null'.
    at handlePromise (D:\Work\Firetrust\Workspace\electron-hideaway\node_modules\execa\index.js:106:10)
    at processTicksAndRejections (internal/process/task_queues.js:89:5)
```
If I `npm i execa@v1.0.0` it no longer throws the exception, when I  `npm i execa@latest` the exception is back. It is also running the command successfully but throws none the less.

If I remove `detached: true` it no longer throws for v2.

**Comment by ehmicky** [comment]
Hi @rynz, that's not good at all. I fixed it in #360. Thanks a lot for reporting this.

**Comment by rynz** [comment]
Thank you!

**Comment by sindresorhus** [comment]
Fixed by #360
